src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -44,4 +43,6 @@ public void setData(Model model) {
     protected void indicateAttemptToExecuteIncorrectCommand() {
         EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent(this));
     }
+    
+    public abstract boolean modifiesData();
```

Comment:
Ugh, not sure why Github wants to mark this as "outdated". Silly github, not sure why we are forced to use it.